sympy/solvers/ode.py

```diff
@@ -4252,19 +4266,30 @@ def _is_special_case_of(soln1, soln2, eq, order, var):
     # means that some value of the constants in sol1 is a special case of
     # sol2 corresponding to a particular choice of the integration constants.
 
+    if isinstance(soln1, Eq):
+        soln1 = soln1.rhs - soln1.lhs
+    if isinstance(soln2, Eq):
+        soln2 = soln2.rhs - soln2.lhs
+
+    # Work for the series solution
+    if soln1.has(Order) and soln2.has(Order):
+        if soln1.getO() == soln2.getO():
+            return _is_special_case_of(soln1.removeO(), soln2.removeO(), eq, order, var)
+    elif soln1.has(Order) or soln2.has(Order):
+        return False
```

Comment:
Everywhere in this function, We need `expr` instead of `Eq` and also for the series solution, We need `expr` and also I am calling `_is_special_case_of ` and during this, I am passing directly `expr.` So maybe the equation will be in the form of `expr` or Eq

Reply:
Looking at it we don't actually need to call `_is_special_case_of` recursively here. We can just do:
```python
if soln1.getO() == soln2.getO():
    return False
else:
    sol1 = sol1.removeO()
    sol2 = sol2.removeO()
```
Then we know that `_is_special_case_of` is always called with `Eq` so we don't need to check `isinstance(Eq)`. The other possibility would be to pass `Eq(0, soln1.removeO())`. Either way I think we should decide what the function is supposed to be called with rather than allowing different types of inputs.